src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs

```diff
@@ -1614,16 +1614,10 @@ public void GetPartsOfConditionalExpression(SyntaxNode node, out SyntaxNode cond
             whenFalse = conditionalExpression.WhenFalse;
         }
 
-        public void GetPartsOfForeachStatement(SyntaxNode statement, out SyntaxToken identifier, out SyntaxNode expression, out IEnumerable<SyntaxNode> statements)
+        public SyntaxNode GetExpressionOfForeachStatement(SyntaxNode statement)
         {
             var commonForeach = (CommonForEachStatementSyntax)statement;
-            identifier = commonForeach is ForEachStatementSyntax { Identifier: var foreachIdentifier }
-                ? foreachIdentifier
-                : default;
-            expression = commonForeach.Expression;
-            statements = commonForeach.Statement is BlockSyntax block
-                ? block.Statements
-                : SpecializedCollections.SingletonEnumerable(commonForeach.Statement);
+            return commonForeach.Expression;
```

Comment:
example of how gnarly this was trying to be a syntaxfact (which i added in the previous PR).  was happy to remove this here.